File: AD-HOC/1867/main.js
function transform(number) {
    while (number.length > 1)
        number = String(number.split('').map(Number).reduce((a, b) => a + b));
    return number;
}

function compare(m, n) {
    if (parseInt(m) > parseInt(n)) return 1;
    if (parseInt(m) < parseInt(n)) return 2;
    return 0;
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
let [m, n] = lines.shift().split(' ');
while (m != '0' || n != '0') {
    console.log(compare(transform(m), transform(n)));
    [m, n] = lines.shift().split(' ');
}

File: AD-HOC/1867/main.dart
import 'dart:io';

String transform(String number) {
  while (number.length > 1) {
    number = number.split('').map(int.parse).reduce((a, b) => a + b).toString();
  }
  return number;
}

int compare(String m, String n) {
  if (int.parse(m) > int.parse(n)) return 1;
  if (int.parse(m) < int.parse(n)) return 2;
  return 0;
}

void main() {
  var [m, n] = stdin.readLineSync()!.split(' ');
  while (m != '0' || n != '0') {
    print(compare(transform(m), transform(n)));
    [m, n] = stdin.readLineSync()!.split(' ');
  }
}
